librad/src/git/trailer.rs

```diff
@@ -0,0 +1,245 @@
+pub mod parser {
+    use super::{Token, Trailer};
+    use nom::{
+        bytes::complete::take_while1,
+        character::complete::{line_ending, not_line_ending, one_of, space0, space1},
+        multi::many0,
+        sequence::{delimited, preceded, separated_pair, terminated},
+        IResult,
+    };
+
+    /// Parse all the possible trailers.
+    /// It stops when it can no longer parse valid trailers.
+    pub fn trailers<'a>(s: &'a str, separators: &'a str) -> IResult<&'a str, Vec<Trailer<'a>>> {
+        many0(|s| trailer(s, separators))(s)
+    }
+
+    /// Parse a trailer, which can have an inlined or multilined value.
+    pub fn trailer<'a>(s: &'a str, separators: &'a str) -> IResult<&'a str, Trailer<'a>> {
+        let parser = separated_pair(token, |s| separator(separators, s), values);
+        let (rest, (token, values)) = parser(s)?;
+        Ok((rest, Trailer { token, values }))
+    }
+
+    /// Parse a trailer token.
+    fn token(s: &str) -> IResult<&str, Token> {
+        take_while1(|c: char| c.is_alphanumeric() || c == '-')(s)
+            .map(|(i, token_str)| (i, Token(token_str)))
+    }
+
+    /// Parse the trailer separator, which can be delimited by spaces.
+    fn separator<'a>(separators: &'a str, s: &'a str) -> IResult<&'a str, char> {
+        delimited(space0, one_of(separators), space0)(s)
+    }
+
+    /// Parse the trailer values, which gathers the value after the separator
+    /// (if any) and possible following multilined values, indented by a
+    /// space.
+    fn values(s: &str) -> IResult<&str, Vec<&str>> {
+        let (r, opt_inline_value) = until_eol(s)?;
+        let (r, mut values) = multiline_values(r)?;
+        if !opt_inline_value.is_empty() {
+            values.insert(0, opt_inline_value)
+        }
+        Ok((r, values))
+    }
+
+    fn multiline_values(s: &str) -> IResult<&str, Vec<&str>> {
+        many0(indented_line_contents)(s)
+    }
+
+    /// Parse an indented line, i.e, a line that starts with a space.
+    /// Extracts the line contents, ignoring the indentation and the
+    /// new line character.
+    fn indented_line_contents(s: &str) -> IResult<&str, &str> {
+        preceded(space1, until_eol)(s)
+    }
+
+    /// Consume the input until the end of the line, ignoring the new line
+    /// character.
+    fn until_eol(s: &str) -> IResult<&str, &str> {
+        terminated(not_line_ending, line_ending)(s)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_message_with_valid_trailers() {
+        let separator = ":";
+        let msg = r#"A subject that succinctly states what this commit does
+
+A longer, more detailed paragraph that describes the what's
+and why's of the changes being introduced by this commit. It
+is usually capped at 72 chars-long lines.
+
+Co-authored-by: John Doe <john.doe@test.com>
+Ticket: #42
+Tested-by:
+    John <john@test.com>
+    Jane <jane@test.com>
+Just-a-token:
```

Comment:
Oh, we can have empty values?!

Reply:
Yap: https://github.com/radicle-dev/radicle-link/pull/271#discussion_r470520559 :brain: